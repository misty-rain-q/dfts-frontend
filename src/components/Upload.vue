<template>
  <div class="upload">
    <a-button @click="turnToUploadPage" type="primary" icon="upload" :size="size">
      上传
    </a-button>
  </div>
</template>

<script>
// @ is an alias to /src
// import HelloWorld from '@/components/HelloWorld.vue';

export default {
  name: 'Upload',
  components: {
  },
  methods: {
    turnToUploadPage() {
      console.log('UploadPage');
    },
  },
};
</script>

<style scoped>

</style>
