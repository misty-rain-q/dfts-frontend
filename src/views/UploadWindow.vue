<template>
    <div class="uploadwindow">
      <div class="left">
        <UploadList class="list"
        :default-file-list="currentFileList"
        ></UploadList>
      </div>

        <div class="right">
          <!-- <click-upload></click-upload> -->
          <ClickUpload
          @filelist-changed="fileListChanged"
          ></ClickUpload>
          <SetFile
          @changeECode = 'cgcode'
          @changeEDate = 'cgdate'
          @changeDset = 'cgds'
          @changeDTime = 'cgdt'
          @changeIdentifyCode = 'cgic'
          @inputCode = 'inputcode'
          @changeExtractionCode="extractionCodeChanged">
          </SetFile>
        </div>

        <!-- <CancelAndUpload></CancelAndUpload> -->

    </div>
</template>

<script>
import Vue from 'vue';

import UploadList from '@/components/UploadList.vue';
import ClickUpload from '@/components/ClickUpload.vue';
import SetFile from '@/components/SetFile.vue';

export default {
  name: 'UploadWindow',
  data() {
    return {
      currentFileList: [],
    };
  },
  methods: {
    fileListChanged(val) {
      this.currentFileList = val;
      // this.currentFileList = this.currentFileList.concat(val);
      this.$emit('file-list-changed', this.currentFileList);
    },
    extractionCodeChanged(val) {
      this.$emit('changeExtractionCode', val);
    },
    // setfile值传递中介
    cgcode(password) {
      this.$emit('changeExtractionCode', password);
    },
    cgdate(datelist) {
      this.$emit('changeEffectiveDate', datelist);
    },
    cgds(downlist) {
      this.$emit('changeDownset', downlist);
    },
    cgdt(downnum) {
      this.$emit('changeDownloadTime', downnum);
    },
    cgic(identifyCode) {
      this.$emit('changeIdentifyCode', identifyCode);
    },
    inputcode(vcode) {
      this.$emit('SendInputCode', vcode);
    },
    clearData() {
      this.currentFileList = [];
    },
  },
  components: {
    UploadList,
    ClickUpload,
    SetFile,
  },
  mounted() {
    Vue.eventBus.$on('upload-window-closed', this.clearData);
  },
};
</script>

<style scoped>
.uploadwindow {
  display: grid;
  height: 100%;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
}

.left {
  margin-left: 5px;
  margin-right: 5px;
  margin-bottom: 5px;
  min-height: 30px;
  max-height: 300px;
}

.list {
  overflow: auto;
}

.right {
  margin-left: 5px;
  margin-right: 5px;
  margin-bottom: 5px;
}

</style>
