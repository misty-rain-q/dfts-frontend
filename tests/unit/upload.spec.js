import { mount } from '@vue/test-utils';
import UploadWindow from '@/views/UploadWindow.vue';
import Vue from 'vue';
import Button from 'ant-design-vue/lib/button';
import Modal from 'ant-design-vue/lib/modal';
import Input from 'ant-design-vue/lib/input';
import CheckBox from 'ant-design-vue/lib/checkbox';
import Upload from 'ant-design-vue/lib/upload';
import Select from 'ant-design-vue/lib/select';
import InputNumber from 'ant-design-vue/lib/input-number';

// Vue.use(ant);
Vue.use(Button);
Vue.use(Modal);
Vue.use(Input);
Vue.use(CheckBox);
Vue.use(Upload);
Vue.use(Select);
Vue.use(InputNumber);

const wrapper = mount(UploadWindow);

describe('UploadWindow.vue', () => {
  it('has inputs', () => {
    const inputs = wrapper.findAll('input');
    console.log(inputs.length);
    expect(inputs.length).toBe(4);
  });
});
