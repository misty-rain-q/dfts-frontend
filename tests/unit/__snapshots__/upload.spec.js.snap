// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadWindow.vue matches snapshot 1`] = `
<div class="uploadwindow">
  <div class="left">
    <div id="iploadlist" class="list"><span class="drag-upload-list"><div class="ant-upload ant-upload-select ant-upload-select-text" style="display: none;"><span role="button" tabindex="0" class="ant-upload"><input type="file" accept="" style="display: none;"></span></div>
    <transition-group-stub tag="div" appear="true" class="ant-upload-list ant-upload-list-text"></transition-group-stub></span>
  </div>
</div>
<div class="right">
  <div class="dragupload"><span><div class="ant-upload ant-upload-drag"><span role="button" tabindex="0" class="ant-upload ant-upload-btn"><input type="file" accept="" multiple="multiple" style="display: none;"><div class="ant-upload-drag-container"><p class="upload-drag-icon"><img src="../assets/file.png" alt="" class="file-icon"></p> <p class="upload-text">
    点击或将文件拖拽到这里上传
  </p> <p class="upload-hint">
    支持扩展名：.rar .zip .doc .docx .pdf .jpg
  </p></div></span></div></span>
</div>
<div id="SetFile"><br>
  <div class="extraction">
    添加提取码：
    <span class="ant-input-affix-wrapper ant-input-password"><input placeholder="不添加则不写" type="password" class="ant-input"><span class="ant-input-suffix"><i aria-label="icon: eye-invisible" tabindex="-1" class="anticon anticon-eye-invisible ant-input-password-icon"><svg viewBox="64 64 896 896" data-icon="eye-invisible" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class=""><path d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"></path><path d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"></path></svg></i></span></span>
  </div> <br>
  <div><span>有效日期：&nbsp;&nbsp;</span>
    <div tabindex="0" class="date ant-select ant-select-enabled" style="width: 120px;">
      <div role="combobox" aria-autocomplete="list" aria-haspopup="true" aria-controls="test-uuid" class="ant-select-selection ant-select-selection--single">
        <div class="ant-select-selection__rendered">
          <div title="1天" class="ant-select-selection-selected-value" style="display: block; opacity: 1;">
            1天
          </div>
        </div><span unselectable="on" class="ant-select-arrow" style="user-select: none;"><i aria-label="icon: down" class="anticon anticon-down ant-select-arrow-icon"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class=""><path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path></svg></i></span>
      </div>
    </div>
  </div> <br> <span>可下载次数：&nbsp;&nbsp;</span>
  <div tabindex="0" class="counter ant-select ant-select-enabled" style="width: 90px;">
    <div role="combobox" aria-autocomplete="list" aria-haspopup="true" aria-controls="test-uuid" class="ant-select-selection ant-select-selection--single">
      <div class="ant-select-selection__rendered">
        <div title="无限次" class="ant-select-selection-selected-value" style="display: block; opacity: 1;">
          无限次
        </div>
      </div><span unselectable="on" class="ant-select-arrow" style="user-select: none;"><i aria-label="icon: down" class="anticon anticon-down ant-select-arrow-icon"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class=""><path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path></svg></i></span>
    </div>
  </div>
  <div class="ant-input-number ant-input-number-disabled">
    <div class="ant-input-number-handler-wrap"><span class="ant-input-number-handler ant-input-number-handler-up " unselectable="unselectable" role="button" aria-label="Increase Value" aria-disabled="true"><i aria-label="icon: up" class="anticon anticon-up ant-input-number-handler-up-inner"><svg viewBox="64 64 896 896" data-icon="up" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class=""><path d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"></path></svg></i></span><span class="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled" unselectable="unselectable" role="button" aria-label="Decrease Value" aria-disabled="true"><i aria-label="icon: down" class="anticon anticon-down ant-input-number-handler-down-inner"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class=""><path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path></svg></i></span></div>
    <div class="ant-input-number-input-wrap"><input role="spinbutton" aria-valuemin="1" aria-valuenow="1" autocomplete="off" disabled="disabled" min="1" step="1" id="inputNumber" class="ant-input-number-input"></div>
  </div>
</div>
</div>
</div>
`;
